src/components/organisms/shared/OakPagination/OakPagination.tsx

```diff
@@ -131,29 +132,40 @@ const OakEllipsis = () => {
 };
 
 export const OakPagination = ({
-  currentPage,
+  initialPage,
   totalPages,
   nextHref,
   prevHref,
   paginationHref,
   pageName,
+  onPageChange,
 }: OakPaginationProps) => {
-  const [activePage, setActivePage] = useState(currentPage);
+  const [activePage, setActivePage] = useState(initialPage);
 
   const pages = generatePageNumbers(activePage, totalPages);
 
   const isFirstPage = activePage <= 1;
   const isLastPage = activePage >= totalPages;
 
-  const handleNumberClick = (num: number) => {
+  const handleNumberClick = (num: number, event: React.MouseEvent) => {
+    event.preventDefault();
     setActivePage(num);
+    onPageChange(num);
   };
 
-  const handleChevronClick = (direction: "backwards" | "forwards") => {
-    setActivePage((currNum) => currNum + (direction === "backwards" ? -1 : 1));
+  const handleChevronClick = (
+    direction: "backwards" | "forwards",
+    event:
+      | React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>
+      | React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>,
```

Comment:
🏄🏻 to save some keystrokes in future this could be `React.UIEvent` which all other events extend. Since you don't really care about what type of event this receives so long as it can be prevented you can leave out the generics

```suggestion
    event: React.UIEvent
```

Reply:
Nice, changed ✅